File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.js
/**
 * @param {number[]} nums
 * @param {number[]} l
 * @param {number[]} r
 * @return {boolean[]}
 */
var checkArithmeticSubarrays = function(nums, l, r) {
    let temp=[]
    let res=[]
    
    for(let i=0;i<l.length;i++){
        let j=0,k=1
        temp=nums.slice(l[i],r[i]+1)
        temp.sort((num1,num2)=>num1-num2)
        console.log(temp)
        if(temp.length===2){
            res.push(true)
            continue
        }
        while(k+1<temp.length){
            if(temp[k]-temp[j]===temp[k+1]-temp[k]){
                j++
                k++
                if(k+1===temp.length){
                    res.push(true)
                    // j=0
                    // k=1
                }
                
            }else{
                res.push(false)
                // j=0
                // k=1
                break
            }    
        }
        
        
    }
    return res
};

File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.py
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        ans=[]
        i=0
        while i in range(len(l)):
            temp=nums[l[i]:r[i]+1]
            temp.sort()
            k=1
            if len(temp)==2:
                ans.append(True)
            while k+1<len(temp):
                if temp[k]-temp[k-1]==temp[k+1]-temp[k]:
                    k+=1
                    if k+1==len(temp):
                        ans.append(True)
                else:
                    ans.append(False)
                    break
            i+=1
        return ans